client/src/pages/reports/EngagementDateFormPartial.js

```diff
@@ -0,0 +1,180 @@
+import { Checkbox } from "@blueprintjs/core"
+import CustomDateInput from "components/CustomDateInput"
+import * as FieldHelper from "components/FieldHelper"
+import { FastField, Field } from "formik"
+import { Report } from "models"
+import moment from "moment"
+import PropTypes from "prop-types"
+import React, { useState } from "react"
+import { Col, ControlLabel, FormGroup, HelpBlock } from "react-bootstrap"
+import Settings from "settings"
+import utils from "utils"
+
+const futureEngagementHint = (
+  <HelpBlock>
+    <span className="text-success">This will create a planned engagement</span>
+  </HelpBlock>
+)
+
+function isStartOfDay(date) {
+  return date && moment(date).isSame(moment(date).startOf("day"))
+}
+
+const EngagementDatePartialFormWithDuration = ({
+  setFieldValue,
+  setFieldTouched,
+  validateFieldDebounced,
+  initialValues,
+  edit,
+  values
+}) => {
+  const [isAllDay, setIsAllDay] = useState(() =>
+    getInitalAllDayState(edit, initialValues)
+  )
+
+  return (
+    <FormGroup>
+      <Col sm={2} componentClass={ControlLabel}>
+        Engagement planning
+      </Col>
+      <Col sm={4} style={{ marginLeft: "15px" }}>
+        <Field
+          name="engagementDate"
+          component={FieldHelper.SpecialField}
+          onChange={value => {
+            const sval = value ? moment(value).startOf("minute").toDate() : null
+            setFieldTouched("engagementDate", true, false) // onBlur doesn't work when selecting a date
+            setFieldValue("engagementDate", sval, true)
+            if (!sval) {
+              setIsAllDay(true)
+            } else if (!isStartOfDay(sval)) {
+              setIsAllDay(false)
+            }
+          }}
+          onBlur={() => setFieldTouched("engagementDate")}
+          widget={
+            <CustomDateInput
+              id="engagementDate"
+              withTime
+              fullWidth
+              allDay={isAllDay}
+            />
+          }
+          vertical
+        >
+          {Report.isFuture(values.engagementDate) && futureEngagementHint}
+        </Field>
+      </Col>
+      <Col sm={3} style={{ marginLeft: "15px" }}>
+        <Field
+          name="duration"
+          label="Duration (minutes)"
+          component={FieldHelper.InputField}
+          inputType="number"
+          onWheelCapture={event => event.currentTarget.blur()} // Prevent scroll action on number input
+          onChange={event => {
+            const safeVal =
+              utils.preventNegativeAndLongDigits(event.target.value, 4) || null
+            setFieldTouched("duration", true, false)
+            setFieldValue("duration", safeVal, false)
+            validateFieldDebounced("duration")
+            setIsAllDay(false)
+          }}
+          vertical
+          disabled={isAllDay}
+        />
+      </Col>
+      <Col
+        sm={2}
```

Comment:
The sum of your Col's should be 9 (12 minus the 3 for the right-hand extra info column); it is 11 now. If you change this one to `sm={1}` and the duration to `sm={2}` it looks slightly better, but the date and duration fields are still too wide:
![Screenshot from 2020-11-26 14-26-14](https://user-images.githubusercontent.com/445238/100356333-5dab5500-2ff3-11eb-926a-d0cd9249221f.png)